### PR TITLE
New module for setting up Avi Pool Group settings.

### PR DESCRIPTION
**Pull request by grastogi23**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

avi.avi_poolgroup
##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = /home/grastogi/.ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

New module for setting up Avi Pool Group settings.

```

```

### CODE REVIEW

**Review by khaltore — APPROVED** [review]
+label shipit

**Comment by ericsysmin** [comment]
+label shipip

**Comment by ericsysmin** [comment]
+label shipip

**Comment by ericsysmin** [comment]
shipit

**Comment by ericsysmin** [comment]
shipit

**Comment by ericsysmin** [comment]
+shipit

**Comment by khaltore** [comment]
shipit

File: lib/ansible/modules/network/avi/avi_poolgroup.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+description:
+    - This module is used to configure PoolGroup object
+    - more examples at U(https://github.com/avinetworks/devops)
+requirements: [ avisdk ]
+version_added: 2.3
+options:
+    state:
+        description:
+            - The state that should be applied on the entity.
+        required: false
```
Comment (dagwieers) [inline lib/ansible/modules/network/avi/avi_poolgroup.py]: If a parameter is not required, you do not have to add `required: false` as that is implicit.
Is is easier to find what is required if these are not listed.

File: lib/ansible/modules/network/avi/avi_poolgroup.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+---
+module: avi_poolgroup
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: Module for setup of PoolGroup Avi RESTful Object
+description:
+    - This module is used to configure PoolGroup object
+    - more examples at U(https://github.com/avinetworks/devops)
+requirements: [ avisdk ]
+version_added: 2.3
```
Comment (dagwieers) [inline lib/ansible/modules/network/avi/avi_poolgroup.py]: This should be string.
```yaml
version_added: "2.3"
```

**Comment by dagwieers** [comment]
It would be nice if new modules have check-mode support.
Please consider adding it.

**Review by dagwieers — CHANGES_REQUESTED** [review]

**Comment by ericsysmin** [comment]
shipit

**Comment by abadger** [comment]
@grastogi23 Just a note here (not blocking the shipits)... you probably want to consider adding a version check.  Since these modules seem to be tightly coupled to the avi sdk, you'll want to be careful that the sdk version that's installed is compatible with the ansible module.

**Comment by gundalow** [comment]
@abadger Good point
@grastogi23 Can you please raise a PR to add that to all the existing modules

**Pull request merged**